### ISSUE TITLE
[rpc context module]: register async methods

### ISSUE THREAD

**Issue by niklasad1** [paritytech/jsonrpsee#291]
It would be nice if we could make it possible for the servers to allow to register `async methods`.

Something like:

```rust
   rpc_module.register_method_async("chain_getHeader", |params, chain| async {
     let hash = params.one()?;     
     chain.header(Some(hash)).await
   })?;
```

**Comment by niklasad1** [comment]
I did some naive attempts on this but couldn't get it work because `async closures` is thing in stable Rust AFAIU

//cc @maciejhirsz

**Comment by niklasad1** [comment]
Closed by #327
